spark/v3.4/spark/src/test/java/org/apache/iceberg/spark/TestSparkV2Filters.java

```diff
@@ -264,4 +288,198 @@ public void testNotIn() {
         Expressions.and(Expressions.notNull("col"), Expressions.notIn("col", 1, 2));
     Assert.assertEquals("Expressions should match", expected.toString(), actual.toString());
   }
+
+  @Test
+  public void testYear() {
+    ScalarFunction<Integer> dateToYear = new YearsFunction.DateToYearsFunction();
+    UserDefinedScalarFunc udf =
+        new UserDefinedScalarFunc(
+            dateToYear.name(),
+            dateToYear.canonicalName(),
+            expressions(FieldReference.apply("col1")));
+    testUDF(udf, Expressions.year("col1"), 2021, DataTypes.IntegerType);
+  }
+
+  @Test
+  public void testMonth() {
+    ScalarFunction<Integer> dateToMonth = new MonthsFunction.DateToMonthsFunction();
+    UserDefinedScalarFunc udf =
+        new UserDefinedScalarFunc(
+            dateToMonth.name(),
+            dateToMonth.canonicalName(),
+            expressions(FieldReference.apply("col1")));
+    testUDF(udf, Expressions.month("col1"), 5, DataTypes.IntegerType);
+  }
+
+  @Test
+  public void testDay() {
+    ScalarFunction<Integer> dateToDay = new DaysFunction.DateToDaysFunction();
+    UserDefinedScalarFunc udf =
+        new UserDefinedScalarFunc(
+            dateToDay.name(), dateToDay.canonicalName(), expressions(FieldReference.apply("col1")));
+    testUDF(udf, Expressions.day("col1"), 20, DataTypes.IntegerType);
```

Comment:
Here as well, we may want to allow comparison with a string date.

Reply:
Updated.